accounts-db/src/tiered_storage/hot.rs

```diff
@@ -317,6 +360,24 @@ pub mod tests {
         assert_eq!(meta.owner_offset(), MAX_HOT_OWNER_OFFSET);
     }
 
+    #[test]
+    fn test_max_hot_account_offset() {
+        HotAccountOffset::new(0).unwrap();
+        HotAccountOffset::new(MAX_HOT_ACCOUNT_OFFSET).unwrap();
+    }
+
+    #[test]
+    #[should_panic(expected = "OffsetOutOfBounds(34359738368, 34359738360)")]
+    fn test_max_hot_account_offset_out_of_bounds() {
+        HotAccountOffset::new(MAX_HOT_ACCOUNT_OFFSET + HOT_ACCOUNT_OFFSET_ALIGNMENT).unwrap();
```

Comment:
I remember it's because some underlying error type (I remember it's IOError) doesn't implement PartialEq or something so that I cannot directly use assert_eq or something.  Let me see if I can workaround it.

Reply:
Thanks for the tip!  Btw, it's `assert_matches!` instead.

Reply:
Aghh!  Got it.  I missed the `assert!`.